converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs

```diff
@@ -100,6 +109,23 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             LogUnhandledExceptionDelegate(_logger, ex);
 
+            if ((ex is FhirException) && ((FhirException)ex).StatusCode == HttpStatusCode.Forbidden)
+            {
+                _telemetryClient.GetMetric(Constants.CastToFhirForbidden).TrackValue(1);
+            }
+            else if ((ex is DicomWebException) && ((DicomWebException)ex).StatusCode == HttpStatusCode.Forbidden)
+            {
+                _telemetryClient.GetMetric(Constants.DicomToCastforbidden).TrackValue(1);
+            }
+            else if (ex is CredentialUnavailableException)
+            {
+                _telemetryClient.GetMetric(Constants.CastMIUnavailable).TrackValue(1);
+            }
+            else
+            {
+                _telemetryClient.GetMetric(Constants.CastingFailedForOtherReasons).TrackValue(1);
+            }
+
             // Any exception in ExecuteAsync will not shutdown application, call hostApplicationLifetime.StopApplication() to force shutdown.
             // Please refer to .net core issue on github for more details: "Exceptions in BackgroundService ExecuteAsync are (sometimes) hidden" https://github.com/dotnet/extensions/issues/2363
             _hostApplicationLifetime.StopApplication();
```

Comment:
Do we need to flush the telemetry client before the application is stopped? I needed to do that for ifx metrics, but not sure if Telemetry Client requires the same thing https://microsofthealth.visualstudio.com/Health/_git/workspace-platform?path=/dicom/app/Microsoft.Health.Cloud.Dicom.Web/Program.cs&version=GBpersonal/t-loemily/sendToEventGrid&line=36&lineEnd=37&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
General practice is it is not required for the continuously running server application. 
https://docs.microsoft.com/en-us/java/api/com.microsoft.applicationinsights.telemetryclient.flush?view=azure-java-stable